physx/src/controller.rs

```diff
@@ -0,0 +1,93 @@
+#![warn(clippy::all)]
+#![warn(rust_2018_idioms)]
+
+use crate::transform::gl_to_px_v3;
+use glam::Vec3;
+use physx_sys::PxControllerFilterCallback;
+use physx_sys::{
+    PxController, PxControllerFilters, PxControllerFilters_new, PxController_getActor,
+    PxController_getPosition, PxController_move_mut, PxController_release_mut,
+    PxController_setPosition_mut, PxExtendedVec3, PxQueryFilterCallback, PxRigidDynamic,
+};
+use std::ptr::null;
+
+pub struct Controller {
+    controller: *mut PxController,
+    filters: PxControllerFilters,
+}
+
+impl Controller {
+    pub fn new(controller: *mut PxController) -> Self {
+        unsafe {
+            let filters = PxControllerFilters_new(
+                null(),
+                null::<PxQueryFilterCallback>() as *mut PxQueryFilterCallback,
+                null::<PxControllerFilterCallback>() as *mut PxControllerFilterCallback,
+            );
+
+            Self {
+                controller,
+                filters,
+            }
+        }
+    }
+
+    pub fn get_raw_mut(&mut self) -> *mut PxController {
```

Comment:
Can we implement this using the `GetRaw` trait instead?

Reply:
👍 Didn't see that